### PR TITLE
add: backend endpoint for the most viewed articles and the http layer

### LINKED ISSUE #179: Create http layer for getting articles
Function that will get articles from endpoint.
Arguments: first index, number of articles
Should return array of objects with properites:
- id
- topic
- dateAdded
- lastEdition
- numberOfViewers (last week)
- numberOfEditions (last week)

Konrad EDIT: I have another vision for this. I would like the backend endpoint to be paginated, so I would like it to return a certain page, given the number of articles per page specified and the starting index of the articles. So the index would possibly be in the middle of the page. 

I also imagine that this function should return more than just the array of objects. It should return the page number as well as the number of the last page that backend can return given the number of articles per page. So an object with a structure like this:
{
 page: ,
 total_pages: ,
 items: [ ],
}

### PR DESCRIPTION
**Pull request by kpagacz**
Closes #179

How to test:
* make sure to undo the seeds to the database and seed it again with the new and updated seeds. In the console:
```
npx sequelize db:seed:undo:all
npx sequelize db:migrate:undo:all
npm run migrate
npm run seed
```
* you can easily test the backend function on the development environment using this:
```js
import { getMostViewed } from "./services/articleService.js";

const res = await getMostViewed(2, 10);
console.dir(res);
```
* testing the function from the frontend is a little bit more tricky. I usually run the backend, add "type" : "module" entry to the frontend/package.json and run a test file:
```js
import { getMostViewedArticles } from "./mostViewedArticles.http.js";

const res = await getMostViewedArticles(1, 10);
console.dir(res);
```

### CODE REVIEW

File: giddy-banana/services/articleService.js [hunk 1]
```diff
@@ -96,4 +98,63 @@ async function updateArticle(title, updatedFields) {
...
+ *   - Article title
+ *   - Date of the views recorded
+ *   - Rank amongst all the article this day
+ *   - Number of views this day.
+ *
+ * @throws {Error} When the database operations failed
+ */
+async function getMostViewed(page, itemsPerPage) {
+  if (page <= 0) throw new InvalidArgumentException("The page argument must be positive");
+  if (itemsPerPage <= 0) throw new InvalidArgumentException("the itemsPerPage argument must be positive");
```
Comment (kpagacz) [inline giddy-banana/services/articleService.js:124]:
```suggestion
  if (itemsPerPage <= 0) throw new InvalidArgumentException("The itemsPerPage argument must be positive");
```

**Comment by kpagacz** [comment]
Would including a link to the wikipedia page would be useful to return as well?

They can actually be constructed from the article title, but hey, we can have a link ready as well since I do the join on articles in the backend already.

**Review by karolkozlo — APPROVED** [review]
Works fine

**Review by karolkozlo — APPROVED** [review]
Works fine

**Pull request merged**